src/util/utility-types.ts

```diff
@@ -42,3 +42,11 @@ export type EitherInclusive<T, U> = EitherExclusive<T, U> | (T & U);
 export type ForwardedRefComponent<T, P> = React.ForwardRefExoticComponent<
   React.PropsWithoutRef<P> & React.RefAttributes<T>
 >;
+
+/**
+ * Utility type used when defining custom render props. Uses template for the
+ * members of the possible
```

Comment:
What does "Uses template for the members of the possible" mean? 🤔

Reply:
Ahh, i forgot to finish the thought, and i keep calling type params "templates". 

`Uses a type param for the members of the possible render prop` or something like that.